src/test/java/org/sasanlabs/service/vulnerability/openRedirect/Http3xxStatusCodeBasedInjectionTest.java

```diff
@@ -0,0 +1,749 @@
+            assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY)).isEqualTo(null);
+            verify(http3xxStatusCodeBasedInjection, times(1))
+                    .getVulnerablePayloadLevel5(requestEntity, "www.somedomain.com");
+        } catch (URISyntaxException | MalformedURLException exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    @Test
+    @DisplayName(
+            "Level 5- test that the returnTo query parameter's value is not added to the Location header when it starts with //")
+    void
+            test_That_ReturnToQueryParameterValue_IsNotAddedToLocationHeader_WhenItStartsWith_DoubleSlash_Level5() {
+        try {
+            String redirectUrl = "//www.somedomain.com";
+            RequestEntity<String> requestEntity =
+                    new RequestEntity<>(
+                            HttpMethod.GET,
+                            new URI("https://somedomain.com?returnTo=//www.somedomain.com"));
+            ResponseEntity<?> responseEntity =
+                    http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel5(
+                            requestEntity, redirectUrl);
+            assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+            assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY)).isEqualTo(null);
+            verify(http3xxStatusCodeBasedInjection, times(1))
+                    .getVulnerablePayloadLevel5(requestEntity, "//www.somedomain.com");
+        } catch (URISyntaxException | MalformedURLException exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    @Test
+    @DisplayName(
+            "Level 5- test that the returnTo query parameter's value is not added to the Location header when it starts with null byte character")
+    void
+            test_That_ReturnToQueryParameterValue_IsNotAddedToLocationHeader_WhenItStartsWith_NullByteCharacter_Level5() {
+        try {
+            String redirectUrl = (char) 0 + "//www.somedomain.com";
+            RequestEntity<String> requestEntity =
+                    new RequestEntity<>(
+                            HttpMethod.GET,
+                            new URI(
+                                    "https://somedomain.com?returnTo=//"
+                                            + (char) 0
+                                            + "www.somedomain.com"));
+            ResponseEntity<?> responseEntity =
+                    http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel5(
+                            requestEntity, redirectUrl);
+            assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+            assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY)).isEqualTo(null);
+            verify(http3xxStatusCodeBasedInjection, times(1))
+                    .getVulnerablePayloadLevel5(requestEntity, (char) 0 + "//www.somedomain.com");
```

Comment:
Try using null character present in Constants file.

Reply:
Okay. Noted